pkg/obfuscate/appsec_test.go

```diff
@@ -0,0 +1,1365 @@
+              ],
+              "highlight": [
+                "highlighted SENSITIVE_VALUE value 1",
+                "highlighted SENSITIVE_VALUE value 2",
+                "highlighted SENSITIVE_VALUE value 3"
+              ],
+              "value": "the entire SENSITIVE_VALUE value"
+            },
+            {
+              "key_path": [
+                0,
+                1,
+                "k1",
+                2,
+                "k3"
+              ],
+              "highlight": [
+                "highlighted value 1",
+                "highlighted value 2",
+                "highlighted value 3"
+              ],
+              "value": "the entire value"
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "id": "crs-x002-210",
+      "name": "Fake AppSec rule",
+      "tags": [ "attack-attempt", "crs-x002-210" ],
+      "rule_matches": [
+        {
+          "parameters": [
+            {
+              "address": "some address",
+              "key_path": [
+                0,
+                1,
+                "k1",
+                2,
+                "SENSITIVE_KEY"
+              ],
+              "highlight": [
+                "highlighted SENSITIVE_VALUE value 1",
+                "highlighted SENSITIVE_VALUE value 2",
+                "highlighted SENSITIVE_VALUE value 3"
+              ],
+              "value": "the entire SENSITIVE_VALUE value"
+            },
+            {
+              "key_path": [
+                0,
+                1,
+                "k1",
+                2,
+                "k3"
+              ],
+              "highlight": [
+                "highlighted SENSITIVE_VALUE value 1",
+                "highlighted SENSITIVE_VALUE value 2",
+                "highlighted SENSITIVE_VALUE value 3"
+              ],
+              "value": "the entire SENSITIVE_VALUE value"
+            },
+            {
+              "key_path": [
+                0,
+                1,
+                "k1",
+                2,
+                "k3"
+              ],
+              "highlight": [
+                "highlighted value 1",
+                "highlighted value 2",
+                "highlighted value 3"
+              ],
+              "value": "the entire value"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+`
+	notMatching := strings.NewReplacer("SENSITIVE_VALUE", "", "SENSITIVE_KEY", "key").Replace(matching)
+
+	for _, bc := range []struct {
+		name, input string
+	}{
+		{
+			name:  "matching",
+			input: matching,
+		},
+		{
+			name:  "not-matching",
+			input: notMatching,
+		},
+	} {
+		b.Run(bc.name, func(b *testing.B) {
+			b.Run("scanner", func(b *testing.B) {
+				o := newAppSecObfuscator(keyRE, valueRE)
+				b.ReportAllocs()
+				b.ResetTimer()
+				for n := 0; n < b.N; n++ {
+					_, err := o.obfuscateAppSec(bc.input)
+					if err != nil {
+						b.Fatal(err)
+					}
+				}
+			})
+
+			b.Run("unmarshalled", func(b *testing.B) {
```

Comment:
Can you please explain what this is? Or what `obfuscatorWithJSONParsing` is?

Reply:
Fixed in https://github.com/DataDog/datadog-agent/pull/11324/commits/9718f7d1bd2bde255789cd2697bf1ce59549b57d